soc/arm/rpi_pico/rp2/soc.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ * Copyright (c) 2021 Yonatan Schachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief System/hardware module for Raspberry Pi RP2040 family processor
+ *
+ * This module provides routines to initialize and support board-level hardware
+ * for the Raspberry Pi RP2040 family processor.
+ */
+
+#include <kernel.h>
+#include <init.h>
+#include <logging/log.h>
+
+
+#ifdef CONFIG_RUNTIME_NMI
+extern void z_arm_nmi_init(void);
+#define NMI_INIT() z_arm_nmi_init()
+#else
+#define NMI_INIT()
+#endif
+
+LOG_MODULE_REGISTER(soc, CONFIG_SOC_LOG_LEVEL);
+
+void rp2_init(void);
```

Comment:
It was because of the KHZ/MHZ collision. It solved the issue by having one file include Zephyr headers, and one include HAL headers. But if the HAL is patched, it's no longer needed.